README.md

```diff
@@ -58,59 +52,72 @@ Before you do so, you should check out our contributing guidelines in the
+- Plenty of disk space (since the builds need to use some heavyweight 
+  full-OS containers in order to build debs and RPMs).
+
+Then, run `make felix-docker-image`, for example, to build the `calico/felix`
+container or `make help` for other options.
 
-We recommend using a Python virtualenv to isolate your dev environment.
-We typically develop on Ubuntu 14.04 (if you're using a later version, 
-make sure you develop with python2.7).  On Ubuntu, to install the dependencies,
-create a virtualenv and install Calico into it:
+## How can I run Felix's unit tests.
 
-    apt-get install git libffi-dev libyajl2 python-dev python-pip
-    pip install coverage tox virtualenv
-    virtualenv env
-    source env/bin/activate
-    pip install -e .
-    
-To run specific unit tests, use `nosetests`; for example:
+After installing the prerequisites above, run `make ut` to run all the 
+tests, `make go-ut` to run Go tests only or `make python-ut` to run 
+Python tests.
 
-    nosetests calico.felix.test.test_selectors
-    
-To deactivate the virtualenv:
+## How can I run Go tests more interactively.
 
-    deactivate
-    
-To run the unit tests (from outside a virtualenv), run:
+If you want to be able to run unit tests for specific packages for more iterative 
+development, you'll need to install 
 
-    ./run-unit-test.sh --develop
-    
-To run felix with log output to screen, first become root (using, 
-for example `sudo -i`), then:
+- go 1.7+
 
-    cd <path to calico dir>
-    source env/bin/activate
-    FELIX_LOGSEVERITYSCREEN=INFO calico-felix
-    
-*Note:* Felix must be run as root because it needs to moanipulate the
-kernel routing table and firewall rules.
+then run `make update-tools` to install ginkgo, which is the test tool used to 
+run Felix's unit tests.
 
-## How do I build/run Felix
+There are several ways to run ginkgo.  One option is to change directory to the 
+package you want to test, then run `ginkgo`.  Another is to use ginkgo's
+watch feature to monitor files for changes:
+```
+cd go
+ginkgo watch -r
+```
+Ginkgo will re-run tests as files are modified and saved.
+
+## How do I build packages/run Felix
 
 ### Docker
 
-Felix can be run inside Docker. See the `docker_build_and_run.sh` script for details on building and running it.
-[![Analytics](https://calico-ga-beacon.appspot.com/UA-52125893-3/calico/README.md?pixel)](https://github.com/igrigorik/ga-beacon)
+After building the docker image (see above), you can run felix with, for example:
+`docker run --privileged -e FELIX_LOGSEVERITYSCREEN=INFO calico/felix`
+
+### Debs and RPMs
+
+The `Makefile` has targets for building debs and RPMS for different platforms.
+By using docker, the build does not need to be run on the target platform.
+```
+make trusty-deb
+make xenial-deb
+make rpm
+```
+The packages (and source packages) are output to the dist directory.
+
+By default, debs are built with a "snapshot" version number that is
+greater than the current release version.
```

Comment:
I don't think I've done that for RPMs yet.  It was needed to make PPA uploads work well so I was forced to do it for debs.